homeassistant/components/flume/sensor.py

```diff
@@ -0,0 +1,246 @@
+        querystring = {"user": "false", "location": "false"}
+        headers = {"authorization": "Bearer " + self._bearer + ""}
+        response = requests.request("GET", url, headers=headers, params=querystring)
+
+        _LOGGER.debug("getDevices Response: %s", response.text)
+
+        if response.status_code == 200:
+            return json.loads(response.text)["data"]
+        else:
+            raise Exception(
+                "getDevices Response Code not Successful. Returned {}".format(
+                    response.status_code
+                )
+            )
+
+
+class FlumeData:
+    """Get the latest data and update the states."""
+
+    def __init__(
+        self,
+        username,
+        password,
+        client_id,
+        client_secret,
+        device_id,
+        time_zone,
+        scan_interval,
+    ):
+        """Initialize the data object."""
+        self._username = username
+        self._password = password
+        self._client_id = client_id
+        self._client_secret = client_secret
+        self._device_id = device_id
+        self._scan_interval = scan_interval
+        self._time_zone = time_zone
+        self.value = None
+
+        flume_auth = FlumeAuth(username, password, client_id, client_secret)
+
+        self._user_id = flume_auth._user_id
+        self._bearer = flume_auth._bearer
+        self.update()
+
+    def update(self):
+        """Return updated value for session."""
+        url = (
+            "https://api.flumetech.com/users/"
+            + str(self._user_id)
+            + "/devices/"
+            + str(self._device_id)
+            + "/query"
+        )
+
+        utc_now = pytz.utc.localize(datetime.utcnow())
+        time_zone_now = utc_now.astimezone(pytz.timezone(self._time_zone))
+
+        since_datetime = (time_zone_now - self._scan_interval).strftime(
+            "%Y-%m-%d %H:%M:00"
+        )
+        until_datetime = time_zone_now.strftime("%Y-%m-%d %H:%M:00")
+
+        queryDict = {
+            "queries": [
+                {
+                    "since_datetime": since_datetime,
+                    "until_datetime": until_datetime,
+                    "bucket": "MIN",
+                    "request_id": "update",
+                    "units": "GALLONS",
+                }
+            ]
+        }
+
+        headers = {"authorization": "Bearer " + self._bearer + ""}
+        response = requests.post(url, json=queryDict, headers=headers)
+
+        _LOGGER.debug("Update URL: %s", url)
+        _LOGGER.debug("Update headers: %s", headers)
+        _LOGGER.debug("Update queryDict: %s", queryDict)
+        _LOGGER.debug("Update Response: %s", response.text)
+
+        if response.status_code == 200:
+            self.value = json.loads(response.text)["data"][0]["update"][0]["value"]
+        else:
+            raise Exception(
+                "getDevices Response Code not Successful. Returned {}".format(
```

Comment:
Exception message should be updated for production.

Something like `Can't update flume data for user id {}. Response code returned : {}`